core/src/test/java/com/google/googlejavaformat/java/FormatterTest.java

```diff
@@ -418,4 +418,113 @@ public void throwsFormatterException() throws Exception {
     } catch (FormatterException expected) {
     }
   }
+
+  @Test
+  public void blankLinesImportComment() throws FormatterException {
+    String withBlank =
+        "package p;\n"
+            + "\n"
+            + "/** test */\n"
+            + "\n"
+            + "import a.A;\n"
+            + "\n"
+            + "class T {\n"
+            + "  A a;\n"
+            + "}\n";
+    String withoutBlank =
+        "package p;\n"
+            + "\n"
+            + "/** test */\n"
+            + "import a.A;\n"
+            + "\n"
+            + "class T {\n"
+            + "  A a;\n"
+            + "}\n";
+
+    // Formatting deletes the blank line between the "javadoc" and the first import.
+    assertThat(new Formatter().formatSource(withBlank)).isEqualTo(withoutBlank);
+    assertThat(new Formatter().formatSourceAndFixImports(withBlank)).isEqualTo(withoutBlank);
+    assertThat(new Formatter().formatSource(withoutBlank)).isEqualTo(withoutBlank);
+    assertThat(new Formatter().formatSourceAndFixImports(withoutBlank)).isEqualTo(withoutBlank);
+
+    // Just fixing imports preserves whitespace around imports.
+    assertThat(RemoveUnusedImports.removeUnusedImports(withBlank)).isEqualTo(withBlank);
+    assertThat(ImportOrderer.reorderImports(withBlank)).isEqualTo(withBlank);
+    assertThat(RemoveUnusedImports.removeUnusedImports(withoutBlank)).isEqualTo(withoutBlank);
+    assertThat(ImportOrderer.reorderImports(withoutBlank)).isEqualTo(withoutBlank);
+  }
+
+  @Test
+  public void indentsIfStatementsWithoutBracesButWithCommentsOnTheSameLine()
+          throws FormatterException {
+    assertThat(
+            new Formatter()
+                    .formatSource(
+                            "public class CommentedTest {\n"
+                                    + "\n"
+                                    + "  public boolean compare(ProglangType other) {\n"
+                                    + "    if (this == other) // comment\n"
+                                    + "    return true;\n"
+                                    + "    return false;\n"
+                                    + "  }\n"
+                                    + "}\n"))
+            .isEqualTo(
+                    "public class CommentedTest {\n"
+                            + "\n"
+                            + "  public boolean compare(ProglangType other) {\n"
+                            + "    if (this == other) // comment\n"
+                            + "        return true;\n"
```

Comment:
Hmm, IIRC, the convention is that most lines of code are indented +2 rather than +4 under google-java-format and the [Google Java Style guide](https://google.github.io/styleguide/javaguide.html#s4.2-block-indentation). Have I missed something here?

Reply:
It's my first time using Github, and I didn't know that my commits to my forked repository are going to show up in this pull request.
But you're right, and I'm almost done, I just have to figure out how to clean up the mess that I've made now.
Should / Can I somehow delete this pull request, fork this repository one more time nice and fresh, and continue where I left, so that I can create a new pull request with one commit?

Reply:
@kitonjicsm 

> It's my first time using Github, and I didn't know that my commits to my forked repository are going to show up in this pull request.

Yes, I believe this occurs because you're pushing commits to the same _branch_ that you used to open this Pull Request (PR for short) with.

In this case, since you're using the branch `master` from your personal fork `kitonjicsm/google-java-format` as the "base" for this PR, all commits you push to `master`, including stuff you may had not intended to be associated with this PR, will be shown here.

To get around this sort of scenario, people usually create a unique branch for each separate PR, as opposed to doing all work on `master` branch.

I hope this helps! If you're still confused or stuck, I'd be very happy to give you further pointers. :)

Reply:
> Should / Can I somehow delete this pull request, fork this repository one more time nice and fresh, and continue where I left, so that I can create a new pull request with one commit?

You usually don't need to go as far as re-forking the repository. It's not clear to me from your commits what commands you used, so it's probably best to do something like the instructions I give below. (My instructions assume that you have enough knowledge to do relatively simple Git tasks. If you're stuck with something or I'm unclear on something, feel free to ask for clarification!)

1. Copy and save your work to a different directory on your computer.
2. Close this PR.
3. Re-create `master` on your local, on-computer Git repo and `master` on your on-GitHub fork `kitonjicsm/google-java-format` to match `google-java-format:master` with the following commands (taken and slightly modified from https://stackoverflow.com/a/9646323/2252930). **Warning:** Only run these commands when you've copied and saved your work elsewhere, as these commands will irreversibly delete and reset everything in both your local Git repo and GitHub fork.
    ```
    git remote add upstream https://github.com/google/google-java-format.git
    git fetch upstream
    git checkout master
    git reset --hard upstream/master  
    git push origin master --force 
    ```
4. Create a new local branch with a name of your choosing (like `formatting-comments` or something).
5. Checkout your new local branch (which I'll assume you've named `formatting-comments`).
6. Copy and paste your work back into your local Git repo.
7. Commit your work locally.
8. Push your new local commit to your on-GitHub fork, specifically branch `kitonjicsm:formatting-comments`.
9. Create a new PR from `kitonjicsm:formatting-comments` (as opposed to `kitonjicsm:master`) as you normally would.

Again, I hope this helps!